### ISSUE TITLE
amazonq: project context: "rejectCodeSuggestion not found" after ctrl-z

### ISSUE THREAD

**Issue by chrisgeo** [aws/aws-toolkit-vscode#5390]
## Steps to reproduce the issue

1. Haven't figure out the pattern
2. Be editing with vim/nvim mode enabled, maybe a linter/formatter or two. Trunk.io can be heavy, especially if both are security scanning (Q was turned off for security)
3. Crashes with the notification (paraphrased) "command aws.amazonq.rejectCodeSuggestion' not found"
4. Some after will find any question complex and out of its domain knowledge 

Point four is likely not related yet amusing that the action of crashing is almost a reset.

## Expected behavior
Stays "alive". Remembers context it was part of in the chat, and doesn't take over the screen with a long running loop.

## System details (run `AWS: About` and/or `Amazon Q: About`)

-   OS: macOS 14.5
-   Visual Studio Code version:
```
Version: 1.91.1
Commit: f1e16e1e6214d7c44d078b1f0607b2388f29d729
Date: 2024-07-09T22:07:46.768Z (2 wks ago)
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 23.5.0
```
-   AWS Toolkit version: N/A don't use bundled
-   Amazon Q version:
```
Amazon Q v1.17.0
Published
2024-04-03, 08:40:29
Last released
2024-07-25, 13:24:54
Last updated
2024-07-26, 11:47:41
Identifier
amazonwebservices.amazon-q-vscode
```

**Comment by jpinkney-aws** [comment]
When it happens again can you provide the Amazon Q logs? 

1. In the vscode [command palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette), run `Amazon Q: View Logs`. This reveals the Output panel with `Amazon Q Logs` selected.
2. Click the gear icon and choose `Debug`, then click it again and choose `Set As Default`.
3. Perform the steps to reproduce the issue.
4. Remove any private info from the logs.

**Comment by chrisgeo** [comment]
@jpinkney-aws Good catch, I was just looking and see nothing noticeable 

## amazon lsp logs
```
Fetched context from cmd/metrics_batch.go, po_metric_events/model.go, po_metric_events/consumer.go, po_metric_events/service.go
Query done in 931ms
Indexing done for 175/354 files
Indexing done for 210/354 files
Indexing done for 245/354 files
Indexing done for 280/354 files
Indexing done for 315/354 files
Indexing done for 350/354 files
Index complete! Take 1580.267597417s. Total 354 files, 2831 chunks
```

Q logs `2024-07-29 04:32:49.257 [debug] codewhisperer: request-id:` all the way down.



Only semi interesting piece:

```
[Warn  - 3:19:01 AM] Unknown tokenizer class "CodeSageTokenizer", attempting to construct from base class.
Device is not connected to power source. Set thread to 1. 
Using number of intra-op threads: 1
Using Cuda(Linux) or DirectML(Windows).
[Warn  - 3:19:01 AM] removing requested execution provider "cuda" from session options because it is not available: backend not found.
[Warn  - 3:19:01 AM] removing requested execution provider "dml" from session options because it is not available: backend not found.
```

**Comment by chrisgeo** [comment]
After another this morning
```
LSP server starts
Loaded model from /Users/<redacted>
[Warn  - 6:51:09 PM] Unknown tokenizer class "CodeSageTokenizer", attempting to construct from base class.
Device is not connected to power source. Set thread to 1. 
Using number of intra-op threads: 1
Using Cuda(Linux) or DirectML(Windows).
[Warn  - 6:51:09 PM] removing requested execution provider "cuda" from session options because it is not available: backend not found.
[Warn  - 6:51:09 PM] removing requested execution provider "dml" from session options because it is not available: backend not found.
Embedding provider initialized.
Validating embedding:s1 0.26636229613723583, s2 0.20119692089068866, s3 0.4491747579475595
Indexing 239 files under rootpath: /Users/<redacted>
Starting index 239 files
Indexing done for 23/239 files
Indexing done for 46/239 files
Indexing done for 69/239 files
Indexing done for 92/239 files
Indexing done for 115/239 files
Indexing done for 138/239 files
Indexing done for 161/239 files
Indexing done for 184/239 files
Indexing done for 207/239 files
Indexing done for 230/239 files
Index complete! Take 104.194211792s. Total 239 files, 403 chunks
Index saved to /Users/<redeacted>




2024-07-31 18:51:03.848 [info] Log level: debug
2024-07-31 18:51:03.848 [debug] User agent: AmazonQ-For-VSCode/1.18.0 Visual-Studio-Code/1.91.1 ClientId/11111111-1111-1111-1111-111111111111
2024-07-31 18:51:03.848 [info] Telemetry product: Amazon Q For VS Code
2024-07-31 18:51:03.848 [debug] Telemetry is disabled
2024-07-31 18:51:03.894 [debug] telemetry cache not found, skipping
2024-07-31 18:51:03.903 [debug] firstStartup: This is not the users first use of the extension. Skipped showing Add Connections page.
2024-07-31 18:51:03.903 [debug] skipping: Settings migration ("aws.codeWhisperer.includeSuggestionsWithCodeReferences" -> "amazonQ.showInlineCodeSuggestionsWithCodeReferences"), (scope: workspaceValue), the old setting is not defined for this scope.
2024-07-31 18:51:03.903 [debug] skipping: Settings migration ("aws.codeWhisperer.includeSuggestionsWithCodeReferences" -> "amazonQ.showInlineCodeSuggestionsWithCodeReferences"), (scope: globalValue), the old setting is not defined for this scope.
2024-07-31 18:51:03.903 [debug] skipping: Settings migration ("aws.codeWhisperer.importRecommendation" -> "amazonQ.importRecommendationForInlineCodeSuggestions"), (scope: workspaceValue), the old setting is not defined for this scope.
2024-07-31 18:51:03.903 [debug] skipping: Settings migration ("aws.codeWhisperer.importRecommendation" -> "amazonQ.importRecommendationForInlineCodeSuggestions"), (scope: globalValue), the old setting is not defined for this scope.
2024-07-31 18:51:03.903 [debug] skipping: Settings migration ("aws.codeWhisperer.shareCodeWhispererContentWithAWS" -> "amazonQ.shareContentWithAWS"), (scope: workspaceValue), the old setting is not defined for this scope.
2024-07-31 18:51:03.903 [debug] skipping: Settings migration ("aws.codeWhisperer.shareCodeWhispererContentWithAWS" -> "amazonQ.shareContentWithAWS"), (scope: globalValue), the latest setting is already defined for this scope.
2024-07-31 18:51:03.903 [debug] codewhisperer: new user login, activating inline tutorial. (autotriggerEnabled=true; inlineState=codewhisperer_learnmore_end)
2024-07-31 18:51:03.940 [debug] SSO token cache: read failed (file not found) key: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:03.940 [info] current client registration id=undefined
    expires at undefined
    key = 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:03.940 [info] auth: Connection is not valid: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:03.940 [info] auth: Updating connection state of 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02 to invalid
2024-07-31 18:51:03.940 [info] codewhisperer: active connection changed
2024-07-31 18:51:03.940 [info] secondaryAuth: connectionId=0a1d3a6b-7cfd-4c1f-81db-249c1452ae02, connectionStatus=invalid
2024-07-31 18:51:03.940 [info] codewhisperer: connection states
    connection isValid=false,
    connection isValidCodewhispererCoreConnection=true,
    connection isExpired=true,
    secondaryAuth isExpired=true,
    connection isUndefined=false
2024-07-31 18:51:04.019 [info] auth: Connection is not valid: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:04.019 [info] auth: Updating connection state of 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02 to invalid
2024-07-31 18:51:04.019 [info] codewhisperer: active connection changed
2024-07-31 18:51:04.035 [debug] command: running "aws.amazonq.refreshConnectionCallback" with arguments: []
2024-07-31 18:51:04.040 [info] auth: Connection is not valid: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:04.040 [info] auth: Updating connection state of 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02 to invalid
2024-07-31 18:51:04.040 [info] initialized 'submitFeedback' command with default feedback id: Amazon Q
2024-07-31 18:51:09.045 [info] LspController: LSP already installed
2024-07-31 18:51:09.472 [info] LspController: LSP activated
2024-07-31 18:51:09.472 [info] LspController: Starting to build vector index of project
2024-07-31 18:51:09.528 [info] LspController: Found 239 files in current project 
2024-07-31 18:51:15.051 [info] auth: Updating connection state of 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02 to authenticating
2024-07-31 18:51:15.051 [info] codewhisperer: active connection changed
2024-07-31 18:51:15.051 [info] codewhisperer: connection changed to authenticating: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:15.054 [debug] auth: SSO registration cache: loaded key: {"startUrl":"https://xxxx.awsapps.com/start","region":"us-east-1","scopes":["codewhisperer:completions","codewhisperer:analysis","codewhisperer:conversations","codewhisperer:transformations","codewhisperer:taskassist"]}
2024-07-31 18:51:15.056 [debug] auth: SSO registration cache: loaded key: {"startUrl":"https://xxxx.awsapps.com/start","region":"us-east-1","scopes":["codewhisperer:completions","codewhisperer:analysis","codewhisperer:conversations","codewhisperer:transformations","codewhisperer:taskassist"]}
2024-07-31 18:51:15.056 [debug] AuthSSOServer: Initialized new auth server.
2024-07-31 18:51:15.560 [info] auth: Connection is not valid: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:15.560 [info] auth: Updating connection state of 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02 to invalid
2024-07-31 18:51:15.560 [info] auth: Connection is not valid: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:15.560 [info] auth: Updating connection state of 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02 to invalid
2024-07-31 18:51:40.954 [debug] API request (oidc.us-east-1.amazonaws.com /token): {
  clientId: '1ZEmJvvl_QDOv7VPzPQ5bXVzLWVhc3QtMQ',
  clientSecret: '[omitted]',
  grantType: 'authorization_code',
  redirectUri: 'http://127.0.0.1:61155/oauth/callback',
  codeVerifier: 'YJ_EaxdKA5cioBjJAyMG846R_jB45RfLL4ebOTJLePI',
  code: 'eyJraWQiOiJrZXktMTU2NDAyODA3OCIsImFsZyI6IkhTMzg0In0.eyJwbGFpbnRleHQiOiJ2c29ZNVZvZkVaSkZHa3lUdHlFUWc2NkdKZDI5eGVvendqT3NITVdSWXlVIiwiZXhwIjoxNzIyNDc3NDAwLCJ0eXBlIjoiYXV0aENvZGUifQ.9YHVTT-ZkLmxfjTWplm-FpwgsNiaEXS8Hy8OQtI69H3pfz5eBFGh5T0Niz0ZLLCt'
}
2024-07-31 18:51:41.037 [error] Unable to find 
2024-07-31 18:51:41.522 [debug] API response (oidc.us-east-1.amazonaws.com /token): {
  '$metadata': {
    httpStatusCode: 200,
    requestId: 'c1fed0f7-a76b-4873-aa2a-16691496f2d3',
    extendedRequestId: undefined,
    cfId: undefined
  },
  accessToken: '[omitted]',
  expiresIn: 3600,
  refreshToken: '[omitted]',
  tokenType: 'Bearer'
}
2024-07-31 18:51:41.522 [debug] be: cleared cache, key: {"identifier":"0a1d3a6b-7cfd-4c1f-81db-249c1452ae02","startUrl":"https://xxxx.awsapps.com/start","scopes":["codewhisperer:completions","codewhisperer:analysis","codewhisperer:conversations","codewhisperer:transformations","codewhisperer:taskassist"],"region":"us-east-1"}, reason: reauth successful
2024-07-31 18:51:41.530 [debug] SSO token cache: saved key: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:41.532 [info] auth: Updating connection state of 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02 to valid
2024-07-31 18:51:41.537 [info] codewhisperer: active connection changed
2024-07-31 18:51:41.537 [info] secondaryAuth: connectionId=0a1d3a6b-7cfd-4c1f-81db-249c1452ae02, connectionStatus=valid
2024-07-31 18:51:41.537 [info] codewhisperer: connection states
    connection isValid=true,
    connection isValidCodewhispererCoreConnection=true,
    connection isExpired=false,
    secondaryAuth isExpired=false,
    connection isUndefined=false
2024-07-31 18:51:41.537 [info] codewhisperer: connection changed to valid: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:41.539 [debug] amazonq: Fetching feature configs
2024-07-31 18:51:41.540 [debug] amazonq: Fetching feature configs
2024-07-31 18:51:41.563 [debug] SSO token cache: loaded key: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:41.563 [info] current client registration id=1ZEmJvvl_QDOv7VPzPQ5bXVzLWVhc3QtMQ
    expires at Sun Oct 27 2024 03:18:17 GMT-0700 (Pacific Daylight Time)
    key = 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:41.563 [info] using codewhispererService service configuration: default
2024-07-31 18:51:41.929 [debug] CodeWhisperer: Current feature configs: {}
2024-07-31 18:51:41.953 [debug] CodeWhisperer: Current feature configs: {}
2024-07-31 18:51:41.970 [debug] amazonq: listAvailableCustomizations API request:
    RequestID: ca8c3b4a-db8c-4a26-835f-c6943dc27d9d,
    Customizations:
2024-07-31 18:51:42.044 [info] auth: Connection is valid: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:42.044 [info] auth: Updating connection state of 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02 to valid
2024-07-31 18:51:42.044 [info] auth: Connection is valid: 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02
2024-07-31 18:51:42.044 [info] auth: Updating connection state of 0a1d3a6b-7cfd-4c1f-81db-249c1452ae02 to valid
2024-07-31 18:51:46.528 [debug] AuthSSOServer: Attempting to close server.
2024-07-31 18:51:46.528 [debug] AuthSSOServer: Server closed successfully.
2024-07-31 18:52:54.346 [debug] LspController: Finish building vector index of project
2024-07-31 18:55:03.570 [debug] codewhisperer: request-id: e599f133-1063-45ab-9533-28a22510890e,
    timestamp(epoch): 1722477303566,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  '',
    line number: 3,
    character location: 0,
    latency: 888.7384169995785 ms.
    Recommendations:
    00: terraform plan -var env=$env -var runner_token="XXXXXXXXXXXXXXXXXXXXXXXXX" -out plan.out
2024-07-31 18:55:03.909 [debug] codewhisperer: sendTelemetryEvent requestID: aa0130fa-42e5-4a9f-9e71-aaee203d5e62
2024-07-31 18:55:07.618 [debug] codewhisperer: request-id: e138e1aa-2c27-4c85-be27-6287c17b221c,
    timestamp(epoch): 1722477307616,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'terrafo',
    line number: 3,
    character location: 7,
    latency: 828.6538750082254 ms.
    Recommendations:
    00: rm plan -var env=$env -out plan.out
2024-07-31 18:55:08.035 [debug] codewhisperer: request-id: 510821ac-e079-4b6d-b274-f882d86de18c,
    timestamp(epoch): 1722477308031,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'terrafo',
    line number: 3,
    character location: 7,
    latency: 412.8313339948654 ms.
    Recommendations:
    00: rm plan -var env=$env -var runner_token="XXXXXXXXXXXXXXXXXXXXXXXXX" -out plan.out
    01: rm plan -var runner_token="XXXXXXXXXXXXXXXXXXXXXXXXX" -var environment=$env -out plan.out
2024-07-31 18:55:12.590 [debug] codewhisperer: sendTelemetryEvent requestID: 87f707ae-44ab-4775-9960-ed6ba2ee6e4e
2024-07-31 18:55:13.081 [debug] codewhisperer: request-id: 7ba839be-ab85-4b56-8725-2d374932368b,
    timestamp(epoch): 1722477313079,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'terraform workwo',
    line number: 3,
    character location: 16,
    latency: 810.9406249970198 ms.
    Recommendations:
    00: space select $env
    01: kspace select $env
2024-07-31 18:55:13.500 [debug] codewhisperer: request-id: 447362c0-9e66-46b4-984a-ba411385dcad,
    timestamp(epoch): 1722477313491,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'terraform workwo',
    line number: 3,
    character location: 16,
    latency: 409.65762500464916 ms.
    Recommendations:
2024-07-31 18:55:13.823 [debug] codewhisperer: sendTelemetryEvent requestID: b8515cb2-d6db-4609-9c85-0c9263b54182
2024-07-31 18:55:16.795 [debug] codewhisperer: request-id: 25a90d96-d75b-410a-b471-e4cd03b95ccc,
    timestamp(epoch): 1722477316793,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'terraform workac',
    line number: 3,
    character location: 16,
    latency: 846.3902499973774 ms.
    Recommendations:
    00: kspace select $env
2024-07-31 18:55:17.133 [debug] codewhisperer: sendTelemetryEvent requestID: 1812aca9-8a1e-4247-8977-f9d468265939
2024-07-31 18:55:17.206 [debug] codewhisperer: request-id: aebd9f16-2adc-4481-9716-3801b0057238,
    timestamp(epoch): 1722477317204,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'terraform workac',
    line number: 3,
    character location: 16,
    latency: 410.0484579950571 ms.
    Recommendations:
    00: select -or-create $env
    01: kspace select -or-create=$env
    02: t apply -var-file="env/$env.tfvars" -auto-approve
2024-07-31 18:55:19.052 [debug] codewhisperer: request-id: c571cb13-d0ff-4514-83bb-a5f3bd1b8af6,
    timestamp(epoch): 1722477319043,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'terraform worksp',
    line number: 3,
    character location: 16,
    latency: 624.5871659964323 ms.
    Recommendations:
    00: ace select $env
2024-07-31 18:55:19.358 [debug] codewhisperer: sendTelemetryEvent requestID: 032fd51f-2082-4558-9b17-43a7c3d1aca4
2024-07-31 18:55:20.081 [debug] codewhisperer: request-id: 580e8001-3171-4acd-8ac3-7be4ef89c38a,
    timestamp(epoch): 1722477320079,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'terraform workspa',
    line number: 3,
    character location: 17,
    latency: 566.2931249886751 ms.
    Recommendations:
    00: ce select $env
2024-07-31 18:55:20.474 [debug] codewhisperer: request-id: ad5fc56d-eeae-4579-a29f-34b6befbffb0,
    timestamp(epoch): 1722477320472,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'terraform workspa',
    line number: 3,
    character location: 17,
    latency: 390.9761669933796 ms.
    Recommendations:
2024-07-31 18:56:10.345 [debug] codewhisperer: sendTelemetryEvent requestID: 77a0d5f2-5eb2-4134-9fbf-b48d22cf191f
2024-07-31 18:56:10.952 [debug] codewhisperer: request-id: 32a88be6-eb43-4350-ac23-8f3206c3f907,
    timestamp(epoch): 1722477370948,
    timezone: ,
    datetime:
    vscode version: '1.91.1',
    extension version: '1.18.0',
    filename: 'terraform-apply.sh',
    left context of line:  'env="${1:',
    line number: 1,
    character location: 9,
    latency: 975.5965410023928 ms.
    Recommendations:
    00: -dev}"
2024-07-31 18:57:02.875 [debug] codewhisperer: sendTelemetryEvent requestID: bff56afa-047f-46a6-96ea-e9768422ebe1
```

**Comment by chrisgeo** [comment]
Grabbed this as soon as it happened.

[LSP server starts.log](https://github.com/user-attachments/files/16449010/LSP.server.starts.log)

**Comment by chrisgeo** [comment]
Found more evidence it's reproducible by using the vim or nvim extension. 

When I attempt to `normaal` from `insert` the switch and cursor become stalled/locked and the notification appears. I normally escape modes while they're a suggestion.

**Comment by jpinkney-aws** [comment]
Just for a bit of extra details, which vim/nvim extension are you using? https://marketplace.visualstudio.com/items?itemName=vscodevim.vim or a different one?

**Comment by chrisgeo** [comment]
@jpinkney-aws That's the one

**Comment by leigaol** [comment]
I was able to reproduce the `aws.amazonq.rejectCodeSuggestion` not found issue by doing below steps:
1. Install Vim
2. Trigger a code suggestion from Amazon Q.
3. Press tab to accept it.
4. Press Ctrl + Z to revoke the acceptance, then you will see the inline suggestion shown again. 
5. Press Esc to reject it, then you will see `aws.amazonq.rejectCodeSuggestion` not found. 

In fact this is reproducible even without Vim. The root cause is the Ctrl + Z reverted the editor states and brought back the inline suggestions while the command override subscriptions were not brought back.

During testing, I reproduced the `crashing` issue using Vim. 
1. Install Vim.
2. Trigger a Q suggestion.
3. Press left or right arrow, then the suggestion does not navigate to next, previous suggestion, instead the cursor moves. After this, I am not getting code suggestions unless after sometime or certain conditions. 
The root cause to this is: Vim override the left right keybindings https://github.com/VSCodeVim/Vim/blob/89cf2d0356966303dc1bac7ce40eaef167d44088/package.json#L123
The keybinding override was somehow ranked as a higher priority item than Q left right navigation buttons. 

Before we have a fix, you can change the keybinding of editor.action.inlineSuggest.showNext to something not like right arrow key to avoid such keybinding conflict.

**Comment by chrisgeo** [comment]
This issue manifested in my workspace before I recall that feature being added

**Comment by justinmk3** [comment]
If you uncheck the "workspace index" setting, does the issue still occur?

**Comment by leigaol** [comment]
Hi @chrisgeo , is it possible for you to try out this build from my above mentioned PR https://d1ihu6zq92vp9p.cloudfront.net/faa1a133-f9a7-466c-9187-6d76adc9face/buildArtifacts.zip ?  I verified the fix on my end. It would be great if you can spend a few minutes to test this out. Thank you!

**Comment by chrisgeo** [comment]
Ensured all GPU Options where unset as well as workspace. Same issue.

[aws.q.log](https://github.com/user-attachments/files/16703110/aws.q.log)
<img width="504" alt="image" src="https://github.com/user-attachments/assets/fd84a37d-25f8-4a66-a34b-e728a024a6ba">
<img width="451" alt="image" src="https://github.com/user-attachments/assets/e869c1d3-c051-4c4a-b11c-ee5dc0cd3882">

**Comment by chrisgeo** [comment]
@leigaol Will give it a try tonight / tomorrow morning.

**Comment by justinmk3** [comment]
Merged attempted fix in https://github.com/aws/aws-toolkit-vscode/pull/5482

Prerelease vsix available here: https://github.com/aws/aws-toolkit-vscode/releases/tag/prerelease

An alternative fix is https://github.com/aws/aws-toolkit-vscode/pull/5474 which works by narrowing the `when` clause of our `left` / `right` overrides.

**Comment by hayemaxi** [comment]
This request was addressed in [AWS Toolkit 3.22.0](https://marketplace.visualstudio.com/items?itemName=AmazonWebServices.aws-toolkit-vscode) and/or [Amazon Q 1.23.0](https://marketplace.visualstudio.com/items?itemName=AmazonWebServices.amazon-q-vscode)!

AWS Toolkit changelog: https://github.com/aws/aws-toolkit-vscode/blob/master/packages/toolkit/CHANGELOG.md
Amazon Q changelog: https://github.com/aws/aws-toolkit-vscode/blob/master/packages/amazonq/CHANGELOG.md
Release artifacts: https://github.com/aws/aws-toolkit-vscode/releases
